src/state-manager/TransactionQueue.ts

```diff
@@ -308,7 +313,25 @@ class TransactionQueue {
           }
           // add the data in
           for (const data of payload.stateList) {
-            this.queueEntryAddData(queueEntry, data)
+            if (
+              configContext.stateManager.collectedDataFix &&
+              configContext.stateManager.rejectSharedDataIfCovered
+            ) {
+              const consensusNodes = this.stateManager.transactionQueue.getConsenusGroupForAccount(
+                data.accountId
+              )
+              const coveredByUs = consensusNodes.map((node) => node.id).includes(Self.id)
```

Comment:
is it possible that account will be covered by node, but node will be OOS on this account ?

Reply:
the old 'broadcast_state'  for non binary is removed in a new PR.
checked binary endpoint:

It is ok for a node to be OOS and still broadcast the state it has.  We have our patcher / and will get some alerting for if there is persistent OOS that is non patchable.    supermajority consensus will be fine even if some nodes are OOS.  this can happen when a node rotates in and we have upgraded the monitor a ton to help us tell the difference between expected vs unexpected patcher behavior